heartbeat/jira

```diff
@@ -0,0 +1,281 @@
+#! /bin/bash
+# 
+####################################################################
+# Description: OCF Resource Agent to manage JIRA software.
+# Author     : Saleh A. (saleh.abbas.saber@gmail.com)
+#
+# License    : WTFPL 2
+#
+#        DO WHAT THE FUCK YOU WANT TO PUBLIC LICENSE 
+#                    Version 2, December 2004 
+#
+# Copyright (C) 2004 Sam Hocevar <sam@hocevar.net> 
+#
+# Everyone is permitted to copy and distribute verbatim or modified 
+# copies of this license document, and changing it is allowed as long 
+# as the name is changed. 
+#
+#            DO WHAT THE FUCK YOU WANT TO PUBLIC LICENSE 
+#   TERMS AND CONDITIONS FOR COPYING, DISTRIBUTION AND MODIFICATION 
+#
+#  0. You just DO WHAT THE FUCK YOU WANT TO.
+# 
+####################################################################
+# Parameters:  
+# OCF_RESKEY_statusurl         : Status URL to monitor JIRA 
+#               (default: http://localhost:8080/status)
+# OCF_RESKEY_java_home         : Java Home
+#               (default: /usr/lib/jvm/jre)
+# OCF_RESKEY_jira_installation : Jira installtion directory
+# OCF_RESKEY_jira_user         : User running Jira software
+#               (by default: jira)
+####################################################################
+
+# Initialization
+# Source ocf-shellfuncs
+: ${OCF_FUNCTIONS_DIR=${OCF_ROOT}/lib/heartbeat}
+. ${OCF_FUNCTIONS_DIR}/ocf-shellfuncs
+
+# Usage
+jira_usage() {
+  cat <<_EOF
+Usage: $0 action
+
+Supported Actions:
+    start       : start jira
+    stop        : stop jira 
+    monitor     : show jira status 
+    meta-data   : show the meta-data
+    validate-all: validate the RA configuration
+_EOF
+}
+
+# Start
+jira_start() {
+    # exit immediately if configuration is not valid
+    jira_validate_all || exit $?
+
+    # if resource is already running, bail out early
+    if jira_monitor; then
+        ocf_log info "Resource is already running"
+        return $OCF_SUCCESS
+    fi
+
+    # Starting Jira
+    waittime=300
+    /bin/su -m $jira_user -c "$jira_installation/bin/startup.sh &> /dev/null"
```

Comment:
There's no reason to use full path for su, and it might cause issues for some users.

Reply:
Removed. Thanks